libs/common/src/common.utils.ts

```
@@ -63,11 +58,56 @@ export function convertUrlToDeepLinkUrl(url: string): string {
+
+    const urlSuffix = agentUrlMap.get(urlFlag);
+
+    if (!urlSuffix) {
+      throw new NotFoundException(ResponseMessages.common.error.invalidUrl);
+    }
+    // Add paramId as a path parameter if provided
+    const resolvedUrlPath = paramId ? urlSuffix.replace('#', paramId) : urlSuffix;
+
+    const url = `${agentEndPoint}${resolvedUrlPath}`;
+
+    console.debug(`Generated agent URL: ${url}`);
```

ESLint

[error] Unexpected console statement.

(no-console)